chain/consensus/filcns/upgrades.go

```diff
@@ -1995,10 +1997,25 @@ func buildUpgradeActorsV12MinerFix(oldBuggyMinerCID, newManifestCID cid.Cid) fun
 		adtStore := store.ActorStore(ctx, stateStore)
 
 		// ensure that the manifest is loaded in the blockstore
+
+		// this loads the "correct" bundle for UpgradeWatermelonFix2Height
 		if err := bundle.LoadBundles(ctx, stateStore, actorstypes.Version12); err != nil {
 			return cid.Undef, xerrors.Errorf("failed to load manifest bundle: %w", err)
 		}
 
+		// this loads the second buggy bundle, for UpgradeWatermelonFixHeight
+		_, ok := build.GetEmbeddedBuiltinActorsBundle(actorstypes.Version12, calibnetv12BuggyBundleSuffix2)
+		if !ok {
+			return cid.Undef, xerrors.Errorf("didn't find buggy calibrationnet bundle")
+		}
+
+		// now confirm we have the one we're migrating to
+		if haveManifest, err := stateStore.Has(ctx, newManifestCID); err != nil {
```

Comment:
How does the ` newManifestCID` get into the state store ?

Reply:
Ah, I have a bug here -- I need to load `calibnetv12BuggyBundleSuffix2`, pushing a commit to fix this. Thank you!

With that fix, the two blocks above should do this:

- `bundle.LoadBundles(ctx, stateStore, actorstypes.Version12)` loads the "correct" bundle, cuz that's what's hard-coded as the bundle to use for v12 actors 
- `build.GetEmbeddedBuiltinActorsBundle(actorstypes.Version12, calibnetv12BuggyBundleSuffix2)`, followed by a call to `LoadBundle` should load the second buggy bundle (the one that's currently live)